src/main/java/com/rabbitmq/http/client/ReactorNettyClient.java

```diff
@@ -130,20 +136,22 @@ private HttpClientRequest disableChunkTransfer(HttpClientRequest request) {
 
     private Mono<HttpClientResponse> doPost(Object body, String... pathSegments) {
         return client.put(uri(pathSegments), request -> Mono.just(request)
-            .map(this::addAuthentication)
+            .transform(this::addAuthorization)
             .map(this::disableChunkTransfer)
             .transform(encode(body)));
     }
 
     private Mono<HttpClientResponse> doDelete(String... pathSegments) {
         return client.delete(uri(pathSegments), request -> Mono.just(request)
-            .map(this::addAuthentication)
+            .transform(this::addAuthorization)
             .flatMap(HttpClientRequest::send)
         );
     }
 
-    private HttpClientRequest addAuthentication(HttpClientRequest request) {
-        return request.addHeader(HttpHeaderNames.AUTHORIZATION, authorizationHeader);
+    private Mono<HttpClientRequest> addAuthorization(Mono<HttpClientRequest> request) {
+        return Mono
+            .zip(request, token)
+            .map(tuple -> tuple.getT1().addHeader(HttpHeaderNames.AUTHORIZATION, tuple.getT2()));
```

Comment:
I guess the `token` is that class property, therefore `this.` would help.
And if it is that, it looks like 
```
.doOnNext(r -> r.addHeader(HttpHeaderNames.AUTHORIZATION, this.token))
```
would enough and much efficient.

Reply:
Oh! I see what you mean with `zip()`. That's because `token` is a `Mono`. 
Well, I think `.block()` especially for cached in-memory value isn't evil here at all.
I just mean that I don't see reason in extra `zip` just for resolving credential on demand.

Reply:
`token` updated to `this.token`.
`token` is a `Mono<String>`, so the point here is to get the token only when needed and potentially a value that can change between calls. The solution you suggest would work only if the token doesn't change once the instance has been created. This is actually the case by default (the `Authorization` header uses username / password digest). Does that make sense?

Reply:
OK, but you use `.cache()` there. I think this will eliminate any your attempts to change credentials at runtime.

Am I missing anything else ?

Reply:
Good. It is now. Thank you!